docs(weave): document `Content`

J2-D2-3PO opened the pull request:
## Description

<!--
Include reference to internal ticket "Fixes WB-NNNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
-->

- Fixes https://wandb.atlassian.net/browse/DOCS-1520

Adds subsection to https://weave-docs.wandb.ai/guides/core-types/media documenting support for `Content`

## Depends on

Feature PR https://github.com/wandb/weave/pull/4787

## Testing

yarn start on local

jamie-rasmussen commented on docs/docs/guides/core-types/media.md:
```diff
@@ -3,7 +3,7 @@ import TabItem from '@theme/TabItem';
 # Logging media
 
-Weave supports logging and displaying video, images, and audio.
+Weave supports logging and displaying video, images, audio, PDFs, and CSVs.
```
We don't need to call out CSVs specifically - unlike PDFs or video/images/audio we don't have special in-app viewing capabilities for CSVs, so that is just an example of our generic File support, where you can pass an arbitrary File in or out of an Op and have the full contents available for download.

(It would be reasonable and not difficult to add an in-app preview of a CSV file in the future, but it's not something we have planned work for.)

jamie-rasmussen commented on docs/docs/guides/core-types/media.md:
```diff
@@ -171,3 +171,118 @@ This audio is logged to Weave and automatically displayed in the UI, along with
+  <TabItem value="python" label="Python" default>
+  
+  ### PDFs
+  You can log and visualize various file types like [PDFs](#pdfs) and [CSVs](#csvs) using the `weave.File` object. This includes PDFs with varying lengths or content. Weave captures file metadata like path, MIME type, and size in traces. You can also [preview](#preview-a-file) or [download](#download-a-file) traced files from the **Traces** tab.
```
You can only preview a `weave.File` if we have built a viewer for that mime type, currently just PDFs.

jamie-rasmussen commented on docs/docs/guides/core-types/media.md:
```diff
@@ -171,3 +171,118 @@ This audio is logged to Weave and automatically displayed in the UI, along with
+   
+  ![Trace view of `accept_file_pdf`](imgs/accept-file-pdf.png)
+
+  You can also work with multi-page PDFs:
```
I don't think we need a separate docs section for multi-page PDFs. I had some separate test cases in the `wf` tool to evaluate performance, and because the code for the viewer has a little bit of special handling for the single page case, basically just hiding next/previous buttons and thumbnail page previews that don't make sense in a single page context.

jamie-rasmussen commented on docs/docs/guides/core-types/media.md:
````diff
@@ -171,3 +171,118 @@ This audio is logged to Weave and automatically displayed in the UI, along with
+  accept_file_pdf_many_pages(return_file_pdf_many_pages())
+  ```
+
+  ### CSVs
````
I would consider restructuring the Files section to first talk about attaching any arbitrary file, perhaps using a CSV as an example, but it could be anything. For arbitrary files you just get the captured metadata in the UI and the ability to download. For PDFs logged with this mechanism specifically we also have added the ability to preview in-app. And we may extend that to future filetypes in the future.

J2-D2-3PO commented:
@jamie-rasmussen thanks for the review. Made some updates based on the feedback:

- Focused on `weave.File `support for arbitrary file types
- Removed the multi-page PDF section
- Clarified that only PDFs currently support in-app preview
- Updated examples and headings
